oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/NodeDocument.java

```diff
@@ -1669,6 +1670,24 @@ SortedMap<Revision, String> getLocalMap(String key) {
         return map;
     }
 
+    /**
+     * Returns name of all the properties on this document
+     * <p>
+     *  Note: property names returned are escaped
+     * <p/>
+     * @return Set of all property names (escaped)
```

Comment:
While saving property names in Mongo, we escape those names that are not supported by Mongo.

e.g. `a.b`. Mongo would store this as a `b` sub-document inside `a`, not as `a.b`.

To avoid such cases we do escape before saving into the database.